### PR TITLE
[ENHANCEMENT] Favorite Collectibles

### PR DESCRIPTION
**Pull request by gantunesr**
# Description

This development simplifies the Collectibles Favorite logic by adding a favorites property to the CollectibleMetadata object in the controllers.

## ⚠️  Blockers 

- ~PR #3380~
- ~PR #3638~

# Tasks

- [x] Remove previous favorite code.
- [x] Add controllers update
- [x] Update tests
- [x] Update snapshots (if necessary)
- [x] Update controllers version

# QA Test Case

1. Upgrade the app from the previous release commit to this version and ensure that the favorites feature behaves as expect.
2. Favorites feature as expected, there should be no changes in the behaviour.

# References

- [MM Controllers PR](https://github.com/MetaMask/controllers/pull/623)

### CODE REVIEW

File: app/components/UI/CollectibleContracts/index.js [hunk 1]
```diff
@@ -102,20 +99,28 @@ const CollectibleContracts = ({
...
+
 		// TO DO: Move this fix to the controllers layer
 		collectibles.forEach((collectible) => {
 			if (shouldUpdateCollectibleMetadata(collectible)) {
 				updateCollectibleMetadata(collectible);
 			}
 		});
+
+		// BACKWARDS COMPATIBILITY: Favorite collectibles
+		const previousFavorites = favorites[selectedAddress]?.[chainId] || [];
```
Comment (Cal-L) [inline app/components/UI/CollectibleContracts/index.js:118]: Move logic to migrations file.

File: app/core/Engine.js [hunk 2]
```diff
@@ -183,6 +183,8 @@ class Engine {
 						assetsContractController.getBalancesInSingleCall.bind(assetsContractController),
 					addTokens: tokensController.addTokens.bind(tokensController),
 					addCollectible: collectiblesController.addCollectible.bind(collectiblesController),
+					updateCollectibleFavoriteStatus: () =>
```
Comment (Cal-L) [inline app/core/Engine.js:186]: This belongs in the CollectibleDetectionController. AssetsDetectionController probably needs to be replaced by `TokenDetectionController` & `CollectibleDetectionController`

**Review by Cal-L — COMMENTED** [review]
Left some comments

File: app/components/UI/CollectibleContracts/index.js (hunk 1 above)
Comment (gantunesr) [reply app/components/UI/CollectibleContracts/index.js]: Done

File: app/core/Engine.js (hunk 2 above)
Comment (gantunesr) [reply app/core/Engine.js]: Fixed in PR 3380

File: package.json [hunk 3]
```diff
@@ -95,7 +95,7 @@
   "dependencies": {
     "@exodus/react-native-payments": "git+https://github.com/MetaMask/react-native-payments.git#dbc8cbbed570892d2fea5e3d183bf243e062c1e5",
     "@metamask/contract-metadata": "^1.30.0",
-    "@metamask/controllers": "^22.0.0",
+    "@metamask/controllers": "git+https://github.com/MetaMask/controllers.git#a8edf9d80496f573911b69d71545918313013f26",
```
Comment (Cal-L) [inline package.json:98]: Since controllers PR is merged, lets update this to use latest release

**Comment by Cal-L** [comment]
Let's clear the conflicts

**Comment by gantunesr** [comment]
@Cal-L addressed your comments in `f69c6e0` and `de37439`

**Review by Cal-L — APPROVED** [review]
LGTM

**Comment by chrisleewilcox** [comment]
Given I have multiple NFT's in my MetaMask mobile wallet
And I have some NFT's as favorite
When I upgrade MetaMask mobile
And I navigate to Wallet
And I tap NFT's tab
Then all my NFT's are still displayed
And all my NFT favorites are still the same before upgrade

**Review by chrisleewilcox — APPROVED** [review]
LGTM

**Comment by chrisleewilcox** [comment]
NFT's images/content is not displaying.  Will write up another issue to deal with it separately.

**Review by chrisleewilcox — APPROVED** [review]
LGMT

**Comment by chrisleewilcox** [comment]
Seeing an error after building and launching app on Android device.
image

https://recordit.co/JnggvKA6mx

Last time I ran into this issue I updated branch with main, rebuild and launch on android device.  Branch is again out of sync with main.  Will re-sync branch with main and try again.

**Comment by chrisleewilcox** [comment]
Does this need to be tested with TestFlight and release apk from Bitrise?

**Comment by chrisleewilcox** [comment]
Conflict needs to be resolved.  Also, we need determine if this can only be tested properly with a TestFlight or Bitrise build.  If this can be tested in another way please advise.

**Comment by cortisiko** [comment]
I will remove the needs QA label on this guy for now. @gantunesr can you handle the merge conflicts and update this branch one last time?

cc @sethkfman

**Comment by gantunesr** [comment]
Stalled

**Pull request closed without merging**